packages/flutter/lib/src/material/bottom_navigation_bar.dart

```diff
@@ -131,6 +131,80 @@ enum BottomNavigationBarType {
 /// ```
 /// {@end-tool}
 ///
+/// {@tool dartpad --template=stateful_widget_material}
+/// This example shows [BottomNavigationBar] used in a [Scaffold] Widget with
+/// different interaction patterns. Tapping on the first [BottomNavigationBarItem]
+/// uses the [ScrollController] to animate the [ListView] to the top. The second
+/// [BottomNavigationBarItem] opens a [DropdownButton].
+///
+///```dart
+/// int _selectedIndex = 0;
```

Comment:
Let's start this off as a non-zero index because it's not meaningful in the bottom navigation bar
```suggestion
/// int _selectedIndex = 1;
```